src/asyncWithLDProvider.tsx

```diff
@@ -38,7 +38,7 @@ export default async function asyncWithLDProvider(config: AsyncProviderConfig) {
   let error: Error;
   let fetchedFlags = {};
 
-  const ldClient = initialize(clientSideID, context, { ...wrapperOptions, ...options });
+  const ldClient = (await config.ldClient) ?? initialize(clientSideID, context, { ...wrapperOptions, ...options });
```

Comment:
config.ldClient can be `undefined`, but awaiting `undefined` will return `undefined`. Then it will be coalesced with the initialize call. So we don't need to check if config.ldClient is set or not.